### PR TITLE
[CryptoAuthLib provider] Implementation of psa_export_public_key operation.

### PR DESCRIPTION
**Pull request by RobertDrazkowskiGL**
Implementation of psa_export_public_key operation.
The changes may seem larger than they actually are because of code layout adjustments:
- contents of file key_operations.rs are moved to key_management.rs (now it looks like in other providers' sources), while the file itself  is removed
- after the above mentioned adjustment, the CAL provider's code needed refactoring in few places

A newer rust-cryptoauthlib is added with support for all known public key export scenarios.

Signed-off-by: Robert Drazkowski <Robert.Drazkowski@globallogic.com>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -41,12 +41,12 @@ picky-asn1-x509 = { version = "0.4.0", optional = true }
...
 libc = "0.2.86"
 anyhow = "1.0.38"
-rust-cryptoauthlib = { version = "0.3.0", optional = true }
+rust-cryptoauthlib = { version = "0.3.1", optional = true }
 prost = { version = "0.7.0", optional = true }
 
 [dev-dependencies]
 rand = { version = "0.8.3", features = ["small_rng"] }
-rust-cryptoauthlib = { version = "0.3.0", features=["software-backend"]}
+rust-cryptoauthlib = { version = "0.3.1", features=["software-backend"]}
```
Comment (hug-dev) [inline Cargo.toml:49]: Just my curiosity but is this needed if `rust-cryptoauthlib` is already a normal dependency?

File: e2e_tests/tests/per_provider/normal_tests/export_public_key.rs [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
 use parsec_client::core::interface::requests::Opcode;
 use parsec_client::core::interface::requests::ResponseStatus;
 use parsec_client::core::interface::requests::Result;
+#[cfg(not(feature = "cryptoauthlib-provider"))]
 use picky_asn1_x509::RSAPublicKey;
 
+#[cfg(not(feature = "cryptoauthlib-provider"))]
 #[test]
-fn export_public_key() -> Result<()> {
+fn export_rsa_public_key() -> Result<()> {
```
Comment (hug-dev) [inline e2e_tests/tests/per_provider/normal_tests/export_public_key.rs:16]: Thanks for renaming those tests 👌

**Review by hug-dev — APPROVED** [review]
Thanks!!

File: Cargo.toml (hunk 1 above)
Comment (RobertDrazkowskiGL) [reply Cargo.toml:49]: We assume the version 0.3.1 is the first supporting both scenarios of exporting public key (first from a slot with a private key, second directly from the slot with public key). Assuming that after 0.3.1 became an official crate it started being preferred anyway, this version bump is probably overzealous.

File: e2e_tests/tests/per_provider/normal_tests/export_public_key.rs (hunk 2 above)
Comment (RobertDrazkowskiGL) [reply e2e_tests/tests/per_provider/normal_tests/export_public_key.rs:16]: I have found this makes debugging much easier.

File: Cargo.toml (hunk 1 above)
Comment (hug-dev) [reply Cargo.toml:49]: Ah sorry I realized my sentence was not clear. I meant: is it needed to have `rust-cryptoauthlib` written as a dependency in both `[dependencies]` and `[dev-dependencies]` sections of the `Cargo.toml` file?

Comment (hug-dev) [reply Cargo.toml:49]: Ah right I missed that! Thanks for the explanation, that makes sense

**Review by ionut-arm — APPROVED** [review]

**Pull request merged**